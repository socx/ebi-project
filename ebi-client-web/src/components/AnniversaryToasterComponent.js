import React from 'react';
import Toast from 'react-bootstrap/Toast';
import ToastContainer from 'react-bootstrap/ToastContainer';


const AnniversaryToasterComponent = (props) => {
  const {
    anniversaryCelebrants,
    birthdayCelebrants,
    isAnniversaryToasterVisible,
    isBirthdayToasterVisible,
    setIsAnniversaryToasterVisible,
    setIsBirthdayToasterVisible
  } = props;
  const birthdayList = birthdayCelebrants && birthdayCelebrants.length &&
    birthdayCelebrants.map((celebrant, index) => {
      return <li key={index}>{celebrant}</li>
    });

  const anniversaryList = anniversaryCelebrants && anniversaryCelebrants.length &&
    anniversaryCelebrants.map((celebrant, index) => {
      return <li key={index}>{celebrant}</li>
    });

  return (
    <>
      <ToastContainer position={'middle-center'}>
        {birthdayCelebrants && birthdayCelebrants.length > 0 &&
          <Toast bg={'dark'}
            onClose={setIsBirthdayToasterVisible}
            show={isBirthdayToasterVisible}
          >
            <Toast.Header>
              <img src='' className="rounded me-2" alt="" />
              <strong className="me-auto">Birthdays this month</strong>
              {/* <small className="text-muted">2 seconds ago</small> */}
            </Toast.Header>
            <Toast.Body className={'text-white'}>
              <p>The following have their birthdays coming up this month:-</p>
              <ul>
                {birthdayList}
              </ul>
            </Toast.Body>
          </Toast>
        }
        {anniversaryCelebrants && anniversaryCelebrants.length > 0 &&
          <Toast bg={'dark'}
            onClose={setIsAnniversaryToasterVisible}
            show={isAnniversaryToasterVisible}
          >
            <Toast.Header>
              <img src='' className="rounded me-2" alt="" />
              <strong className="me-auto">Wedding anniversaries this month</strong>
              {/* <small className="text-muted">2 seconds ago</small> */}
            </Toast.Header>
            <Toast.Body className={'text-white'}>
              <p>The following have their anniversaries coming up this month:-</p>
              <ul>
                {anniversaryList}
              </ul>
            </Toast.Body>
          </Toast>
        }
      </ToastContainer>
    </>
  )
}

export default AnniversaryToasterComponent;
