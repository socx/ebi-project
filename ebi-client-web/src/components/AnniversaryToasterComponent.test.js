import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import AnniversaryToasterComponent from './AnniversaryToasterComponent';

describe('AnniversaryToasterComponent', () => {

  const birthdayCelebrants = [
    '1st March - Joe Bloggs',
    '1th March - Peter Pan'
  ];

  test('renders the component', () => {
    const props = {
      birthdayCelebrants: birthdayCelebrants,
      isBirthdayToasterVisible: true,
      setIsBirthdayToasterVisible: ()=> {}
    };
    const container = render(<AnniversaryToasterComponent {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
