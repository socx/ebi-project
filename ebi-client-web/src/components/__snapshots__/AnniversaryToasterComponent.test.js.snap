// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnniversaryToasterComponent renders the component 1`] = `undefined`;
